### PR TITLE
Upgrade to jakarta data 1.0.0 m3

### PR DESCRIPTION
**Pull request by dearrudam**
## Changes

- Bump up the actions that use Node.js 20 runtime (For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/):
  - actions/checkout@v3 to v4;
  - actions/setup-java@v3 to v4;

- Upgrade the API to Jakarta Data 1.0.0-M3

### CODE REVIEW

**Comment by genie-jnosql** [comment]
Can one of the admins verify this patch?

**Comment by dearrudam** [comment]
@otaviojava could you review it, please?

File: jnosql-lite/mapping-lite-column-test/src/main/java/org/eclipse/jnosql/lite/mapping/entities/PersonRepository.java [hunk 1]
```diff
@@ -28,11 +28,11 @@
 import java.util.List;
 
 @Repository
-public interface PersonRepository extends PageableRepository<Person, Long> {
+public interface PersonRepository extends BasicRepository<Person, Long> {
```
Comment (otaviojava) [inline jnosql-lite/mapping-lite-column-test/src/main/java/org/eclipse/jnosql/lite/mapping/entities/PersonRepository.java:31]: Max, please keep the test and use the NoSQL repository instead.

**Comment by dearrudam** [comment]
@otaviojava, could you review it, please?!

File: jnosql-lite/mapping-lite-column-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonCrudRepositoryTest.java [hunk 2]
```diff
@@ -108,13 +108,6 @@ void shouldDeleteEntity() {
         verify(template, times(1)).delete(eq(Person.class),eq(person.getId()));
     }
 
-    @Test
-    void shouldDeleteAllEntities() {
```
Comment (otaviojava) [inline jnosql-lite/mapping-lite-column-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonCrudRepositoryTest.java:112]: could you return this method?

File: jnosql-lite/mapping-lite-column-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonRepositoryTest.java [hunk 3]
```diff
@@ -196,8 +206,8 @@ void shouldReturnFalseIfEntityDoesNotExistById() {
 
 
     @Test
-    void shouldFindAllEntitiesWithPageable() {
-        Pageable pageable = mock(Pageable.class);
+    void shouldFindAllEntitiesWithPageRequest() {
+        PageRequest pageable = mock(PageRequest.class);
```
Comment (otaviojava) [inline jnosql-lite/mapping-lite-column-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonRepositoryTest.java:210]: Could you also rename the variable name?

```suggestion
        PageRequest pageRequest = mock(PageRequest.class);
```

File: jnosql-lite/mapping-lite-document-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonCrudRepositoryTest.java [hunk 4]
```diff
@@ -111,13 +108,6 @@ void shouldDeleteEntity() {
         verify(template, times(1)).delete(eq(Person.class),eq(person.getId()));
     }
 
-    @Test
-    void shouldDeleteAllEntities() {
```
Comment (otaviojava) [inline jnosql-lite/mapping-lite-document-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonCrudRepositoryTest.java:115]: could you return this method?

File: jnosql-lite/mapping-lite-document-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonRepositoryTest.java [hunk 5]
```diff
@@ -206,8 +206,8 @@ void shouldReturnFalseIfEntityDoesNotExistById() {
 
 
     @Test
-    void shouldFindAllEntitiesWithPageable() {
-        Pageable pageable = mock(Pageable.class);
+    void shouldFindAllEntitiesWithPageRequest() {
+        PageRequest pageable = mock(PageRequest.class);
```
Comment (otaviojava) [inline jnosql-lite/mapping-lite-document-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonRepositoryTest.java:210]: the variable name here

File: jnosql-lite/mapping-lite-document-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonRepositoryTest.java [hunk 6]
```diff
@@ -297,9 +297,9 @@ void shouldDeleteByName(){
     }
 
     @Test
-    void shouldFindPageable(){
+    void shouldFindPageRequest(){
         when(template.select(any(DocumentQuery.class))).thenReturn( Stream.of(new Person(), new Person()));
-        Pageable pageable = Pageable.ofPage(10).sortBy(Sort.asc("name"));
+        PageRequest<Person> pageable = PageRequest.of(Person.class).page(10).sortBy(Sort.asc("name"));
```
Comment (otaviojava) [inline jnosql-lite/mapping-lite-document-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonRepositoryTest.java:302]: the variable name here

File: jnosql-lite/mapping-lite-graph-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonRepositoryLiteGraphTest.java [hunk 7]
```diff
@@ -143,8 +145,8 @@ void shouldFindAllEntities() {
     }
 
     @Test
-    void shouldFindAllEntitiesWithPageable() {
-        Pageable pageable = Pageable.ofPage(1);
+    void shouldFindAllEntitiesWithPageRequest() {
+        PageRequest pageable = PageRequest.ofPage(1);
```
Comment (otaviojava) [inline jnosql-lite/mapping-lite-graph-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonRepositoryLiteGraphTest.java:149]: the variable name here

File: jnosql-lite/mapping-lite-processor/src/main/java/org/eclipse/jnosql/lite/mapping/repository/MethodMetadata.java [hunk 8]
```diff
@@ -90,7 +90,7 @@ public String getReturnType() {
 
     public String getParametersSignature() {
         return parameters.stream().map(Parameter::parameterName)
-                .collect(joining(","));
+                .collect(joining(", \n" + " ".repeat(30)));
```
Comment (otaviojava) [inline jnosql-lite/mapping-lite-processor/src/main/java/org/eclipse/jnosql/lite/mapping/repository/MethodMetadata.java:93]: Could you create a constant at 30?

File: jnosql-lite/mapping-lite-processor/src/main/java/org/eclipse/jnosql/lite/mapping/repository/MethodMetadata.java [hunk 9]
```diff
@@ -147,10 +147,10 @@ public String getEntityType() {
         return entityType;
     }
 
-    public Optional<Parameter> findPageable(){
+    public Optional<Parameter> findPageRequest(){
         for (Parameter parameter : this.parameters) {
             TypeElement element = parameter.type();
-            if("jakarta.data.page.Pageable".equals(element.getQualifiedName().toString())){
+            if(PageRequest.class.getName().equals(element.getQualifiedName().toString())){
```
Comment (otaviojava) [inline jnosql-lite/mapping-lite-processor/src/main/java/org/eclipse/jnosql/lite/mapping/repository/MethodMetadata.java:153]: Wey better, thanks Max.

File: jnosql-lite/mapping-lite-processor/src/main/java/org/eclipse/jnosql/lite/mapping/repository/RepositoryUtil.java [hunk 10]
```diff
@@ -29,9 +30,10 @@
 
 final class RepositoryUtil {
 
-    static final Predicate<String> IS_PAGEABLE_REPOSITORY = q -> PageableRepository.class.getName().equals(q);
+    static final Predicate<String> IS_NOSQL_REPOSITORY = q -> NoSQLRepository.class.getName().equals(q);
     static final Predicate<String> IS_CRUD_REPOSITORY = q -> CrudRepository.class.getName().equals(q);
-    static final Predicate<String> IS_JAKARTA_DATA_REPOSITORY = IS_PAGEABLE_REPOSITORY.or(IS_CRUD_REPOSITORY);
+    static final Predicate<String> IS_BASIC_REPOSITORY = q -> BasicRepository.class.getName().equals(q);
```
Comment (otaviojava) [inline jnosql-lite/mapping-lite-processor/src/main/java/org/eclipse/jnosql/lite/mapping/repository/RepositoryUtil.java:35]: Nice, killing a bug 🔪 🐞

**Review by otaviojava — APPROVED** [review]
After those suggestions you can merge it.

File: jnosql-lite/mapping-lite-processor/src/main/java/org/eclipse/jnosql/lite/mapping/repository/MethodMetadata.java (hunk 9 above)
Comment (dearrudam) [reply jnosql-lite/mapping-lite-processor/src/main/java/org/eclipse/jnosql/lite/mapping/repository/MethodMetadata.java:153]: I think when Class-File API comes - https://openjdk.org/jeps/457 - we can do something much better than mustache templates...I guess heheh

File: jnosql-lite/mapping-lite-processor/src/main/java/org/eclipse/jnosql/lite/mapping/repository/MethodMetadata.java (hunk 8 above)
Comment (dearrudam) [reply jnosql-lite/mapping-lite-processor/src/main/java/org/eclipse/jnosql/lite/mapping/repository/MethodMetadata.java:93]: Yeah... my bad! I'll fix it!!!

File: jnosql-lite/mapping-lite-column-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonCrudRepositoryTest.java (hunk 2 above)
Comment (dearrudam) [reply jnosql-lite/mapping-lite-column-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonCrudRepositoryTest.java:112]: The PersonCrudRepository is a `CrudRepository` specialization... and `CrudRepository` doesn't have `deleteAll()` method. Should I make `PersonCrudRepository` a specialization of `NoSQLRepository`?

File: jnosql-lite/mapping-lite-document-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonCrudRepositoryTest.java (hunk 4 above)
Comment (dearrudam) [reply jnosql-lite/mapping-lite-document-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonCrudRepositoryTest.java:115]: The PersonCrudRepository is a `CrudRepository` specialization... and `CrudRepository` doesn't have `deleteAll()` method. Should I make `PersonCrudRepository` a specialization of `NoSQLRepository`?

Comment (otaviojava) [reply jnosql-lite/mapping-lite-document-test/src/test/java/org/eclipse/jnosql/lite/mapping/entities/PersonCrudRepositoryTest.java:115]: Nope, thus, let keep this way.

**Pull request merged**